tests/test_curobo.py

```diff
@@ -26,93 +37,427 @@ def test_curobo():
+                    if bypass_physics:
+                        print(f"Teleporting waypoint {i}/{len(q_traj)}")
+                        robot.set_joint_positions(q)
+                        robot.keep_still()
+                        og.sim.step()
+                        for contact in robot.contact_list():
+                            assert contact.body0 in robot.link_prim_paths
+                            if (
+                                contact.body1 in floor_plane_prim_paths
+                                and contact.body0 in floor_touching_base_link_prim_paths
+                            ):
+                                continue
+                            if th.tensor(list(contact.impulse)).norm() == 0:
+                                continue
+                            print(f"Unexpected contact pair during traj rollout: {contact.body0}, {contact.body1}")
+                            assert (
+                                False
+                            ), f"Unexpected contact pair during traj rollout: {contact.body0}, {contact.body1}"
+                    else:
+                        # Convert target joint positions to command
+                        q = q.cpu()
+                        command = []
+                        for controller in robot.controllers.values():
+                            command.append(q[controller.dof_idx])
+                        command = th.cat(command, dim=0)
+                        assert command.shape[0] == robot.action_dim
+
+                        num_repeat = 3
+                        for j in range(num_repeat):
+                            print(f"Executing waypoint {i}/{len(q_traj)}, step {j}")
+                            env.step(command)
+
+                            for contact in robot.contact_list():
+                                assert contact.body0 in robot.link_prim_paths
+                                if (
+                                    contact.body1 in floor_plane_prim_paths
+                                    and contact.body0 in floor_touching_base_link_prim_paths
+                                ):
+                                    continue
+                                if th.tensor(list(contact.impulse)).norm() == 0:
+                                    continue
+                                print(f"Unexpected contact pair during traj rollout: {contact.body0}, {contact.body1}")
+                                # Controller is not perfect, so collisions might happen
+                                # assert False, f"Unexpected contact pair during traj rollout: {contact.body0}, {contact.body1}"
+
+                            cur_joint_positions = robot.get_joint_positions()
+                            if ((cur_joint_positions - q).abs() < error_tol).all():
+                                break
+
+        og.clear()
+
+        del cmg
+        gc.collect()
+        th.cuda.empty_cache()
+
+    og.shutdown()
```

Comment:
should this be here? or is this guaranteed to be the final test that's executed?